artifacts/mediaTypes.md

```diff
@@ -0,0 +1,211 @@
+- Artifact authors MAY provide new schemas for the config.json object. 
+- Artifacts and tooling that use existing `config.mediaTypes` MUST conform to the artifacts spec.
+- Tools MAY choose to support new `config.mediaTypes`.
+- Tools MUST be able to ignore `config.mediaTypes` they don't support. 
+- Distribution implementations MAY choose to support new `config.mediaTypes`. See the [distribution-spec] for more details on processing of the `config.mediaTypes`
+
+### mediaTypes formatting
+
+New mediaTypes MUST use the following convention:
+
+`application/vnd.`[governingOrg/vendor]`.`[artifactName]`.config.`[version]`+json` 
+
+**Example:** 
+The Helm 3 artifact may be represented as: `application/vnd.cncf.helm/config.v3+json`
+
+## Config Parsing
+
+New artifacts MUST choose from 3 options for how the `config` object is processed and parsed:
+
+- Support the [`application/vnd.oci.image.config.v1+json`](../config.md) schema
+- Define a new schema, which artifact specific tools would need to support
+- Define a `null` schema, where the artifact only uses the `config.mediaType` to identify it's unique artifact type. 
+
+Artifact authors MAY publicize their schema in [./mediaTypeMappings.json]() through a pull request.
+
+Artifact authors understand supporting additional artifact types is optional, and distribution parsing of new config schemas is optional. However, new config schemas may be useful to that artifact tooling experience. 
+
+## Layer mediaTypes
+
+While OCI container images have ordinal layers, supporting overlaying of a unified file system, new artifact authors MAY define how they persist their layers, which may be individual files, or collections of files. The tooling specific to that artifact type owns the processing of the files and how they are extracted on the client. 
+
+The [layer mediaType descriptor in the OCI Image spec](https://github.com/opencontainers/image-spec/blob/master/manifest.md#image-manifest-property-descriptions) identifies the `mediaType` MUST support one of the existing layer formats. 
+
+- **`layers`** *array of objects*
+
+    Each item in the array MUST be a [descriptor](../descriptor.md).
+    
+    Beyond the [descriptor requirements](../descriptor.md#properties), the value has the following additional restrictions:
+
+    - **`mediaType`** *string*
+
+        This [descriptor property](descriptor.md#properties) has additional restrictions for `layers[]`.
+        Implementations MUST support at least the following media types:
+
+        - [`application/vnd.oci.image.layer.v1.tar`](layer.md)
+        - [`application/vnd.oci.image.layer.v1.tar+gzip`](layer.md#gzip-media-types)
+    
+Artifact authors can extend these mediaTypes to identify their unique usage, using the following format: 
+
+`application/vnd.`[governingOrg/vendor]`.`[artifactName]`.layer.`[version][`.tar`|`+tar+gzip`]
+
+**Helm Example**
+
+`application/vnd.cncf.helm.chart.layer.v3.tar`
+`application/vnd.cncf.helm.chart.values.v3.tar`
+
```

Comment:
Just curious - who maintains these namespaces? What does vnd mean?

Reply:
We've discussed a "clearing house" where artifact authors can submit their artifacts for others to consume. This is in the ./artifacts/mediaTypeMappings.json file.  Registry authors could choose to keep these unique to them as well.

Reply:
okay that makes sense. And validation of any media type is out of scope for the spec, and is responsibility of a client or registry?

Reply:
Correct. I figured I'd cover that in the distribution spec.

Reply:
@vsoch  `vnd` stands for `vendor` and it represents a third-party mediatype registered with IANA.

Reply:
Oh interesting - first I was looking around IANA trying to find something container related:

 - [event names](https://www.iana.org/assignments/message-store-events/message-store-events.xhtml#message-store-events-2)
 - [event parameters](https://www.iana.org/assignments/message-store-events/message-store-events.xhtml#message-store-events-4)
 - [MQTP](https://www.iana.org/assignments/mtqp-options/mtqp-options.xhtml#mtqp-options-2)

but actually the better description is on [wikipedia](https://en.wikipedia.org/wiki/Media_type) that explains what you said above - vendor specific mime types. Then I found [this listing](https://www.iana.org/assignments/media-types/media-types.xhtml) but why don't we see anything from Docker there? Is it not registered?

Reply:
People use `vnd` and don't register. I know docker doesn't have their mediatypes registered but coreos has one for etcd I think.

Reply:
Interesting. I guess we can make up whatever we like then :wink:

Reply:
Haha yeah basically.

Reply:
I'm hopeful most artifact owners will use the clearing house/PR approach to broadcast their types. I imagine most registries would limit artifact types to whitelisting known types. Otherwise, we could only present UNKOWN, or show the `mediaType` which would be sorta ugly. I'm hopeful we can make this easy enough for registries to import the script with minimal internal configuration. But, notice I keep saying hopeful. Only time will tell.